### PR TITLE
Remove stale diff reasons.

### PR DESCRIPTION
**Pull request by dragomirtitian**

### CODE REVIEW

File: src/testRunner/compilerRunner.ts [hunk 1]
```diff
@@ -658,11 +662,15 @@ class IsolatedDeclarationTest extends CompilerTestBase {
         );
     }
 
+    verifyDiffReason() {
+        if (this.isOutputMapEquivalent && this.isOutputEquivalent && this.isDiagnosticEquivalent) {
+            ts.Debug.assert(this.diffReason === undefined, "Should not have a diff reason if everything is equivalent");
+        }else {
```
Comment (h-joo) [inline src/testRunner/compilerRunner.ts]: nit: please add a space in between `}` and `else`

File: src/testRunner/compilerRunner.ts [hunk 2]
```diff
@@ -480,13 +482,15 @@ class IsolatedDeclarationTest extends CompilerTestBase {
...
 
-    static transformEnvironment(compilerEnvironment: CompilerTestEnvironment): CompilerTestEnvironment | undefined {
+    static transformEnvironment(compilerEnvironment: CompilerTestEnvironment, shouldNotExclude = !!compilerEnvironment.testCaseContent.settings.isolatedDeclarationDiffReason): CompilerTestEnvironment | undefined {
         const options = compilerEnvironment.compilerOptions;
-        // Exclude tests some tests
-        // - those explicitly not opting into isolatedDeclarations
-        // - those that do not usually emit output anyway
-        if (options.isolatedDeclarations === false || options.noEmit || options.noTypesAndSymbols || !options.declaration) {
-            return undefined;
+        if (!shouldNotExclude) {
```
Comment (h-joo) [inline src/testRunner/compilerRunner.ts:487]: nit: Do we want to merge the nested if statments? It seems like you were trying to do that by covering the statements with brackets.

**Review by h-joo — APPROVED** [review]

File: src/testRunner/compilerRunner.ts (hunk 1 above)
Comment (dragomirtitian) [reply src/testRunner/compilerRunner.ts]: forgot to format. thanks.

File: src/testRunner/compilerRunner.ts (hunk 2 above)
Comment (dragomirtitian) [reply src/testRunner/compilerRunner.ts:487]: Yeah. I did initially do that, but decided against it. The condition just looked complicated and had to follow. I removed the extra `()`

**Pull request merged**